scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -923,31 +991,32 @@ private Windows narrowGlobalConstraints(
    * @return the instance of the activity (if successful; else, an empty object) wrapped as an Optional.
    */
   public @NotNull Optional<SchedulingActivityDirective> createOneActivity(
-      final ActivityExpression activityExpression,
+      final MissingActivityTemplateConflict missingConflict,
       final String name,
       final Windows windows,
-      final EvaluationEnvironment evaluationEnvironment
-  ) throws SchedulingInterruptedException {
+      final EvaluationEnvironment evaluationEnvironment) throws SchedulingInterruptedException
+  {
     //REVIEW: how to properly export any flexibility to instance?
     logger.info("Trying to create one activity, will loop through possible windows");
     for (var window : windows.iterateEqualTo(true)) {
       logger.info("Trying in window " + window);
-      var activity = instantiateActivity(activityExpression, name, window, evaluationEnvironment);
+      var activity = instantiateActivity(missingConflict, name, window, missingConflict.getEvaluationEnvironment());
       if (activity.isPresent()) {
-        return activity;
+          return activity;
       }
     }
     return Optional.empty();
   }
   private Optional<SchedulingActivityDirective> instantiateActivity(
-      final ActivityExpression activityExpression,
+      final MissingActivityTemplateConflict missingConflict,
```

Comment:
why do you pass the conflict if below you are only using the template ?

Reply:
Right. Fixed